src/auth/auth.service.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common'
+import { JwtService } from '@nestjs/jwt'
+import { User } from 'src/users/entities/user.entity'
+import * as bcrypt from 'bcrypt'
+import { InjectRepository } from '@nestjs/typeorm'
+import { Repository } from 'typeorm'
+import { LoginUserDto } from './dto/login-user.dto'
+@Injectable()
+export class AuthService {
+  constructor(
+    @InjectRepository(User)
+    private userRepository: Repository<User>,
+    private jwtService: JwtService,
+  ) {}
+
+  async validateUser(
+    data: LoginUserDto,
+  ): Promise<Omit<User, 'password'>> {
+    const user = await this.userRepository.findOne({
+      where: { email: data.email },
+    })
+    if (!user) {
+      throw new NotFoundException("User doesn't exist")
+    }
+    if (await bcrypt.compare(data.password, user.password)) {
+      const { password, ...userWithNoPassword } = user
+      return userWithNoPassword
+    } else {
+      throw new UnauthorizedException("Email and password don't match")
+    }
+  }
+  async login(data: LoginUserDto): Promise<{
+    user: Partial<User>
+    access_token: string
+  }> {
+
+    const user = await this.validateUser(data)
+
+    const payload = {
+      email: user?.email,
+      role: user?.roles,
+      id: user?.id,
+    }
+    if (!user?.id) {
```

Comment:
this if condition is redundant since the validateUser will not proceed (will an error) if the user is invalid or empty